src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/editdocs/EditDocsAuditLogHandler.java

```diff
@@ -0,0 +1,88 @@
+import static uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCaseFieldDefinition.ADDITIONAL_EVIDENCE_DOCUMENTS;
+import static uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCaseFieldDefinition.DRAFT_DECISION_AND_REASONS_DOCUMENTS;
+import static uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCaseFieldDefinition.FINAL_DECISION_AND_REASONS_DOCUMENTS;
+import static uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCaseFieldDefinition.HEARING_DOCUMENTS;
+import static uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCaseFieldDefinition.HEARING_RECORDING_DOCUMENTS;
+import static uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCaseFieldDefinition.LEGAL_REPRESENTATIVE_DOCUMENTS;
+import static uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCaseFieldDefinition.RESPONDENT_DOCUMENTS;
+import static uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCaseFieldDefinition.TRIBUNAL_DOCUMENTS;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.iacaseapi.domain.UserDetailsProvider;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCase;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCaseFieldDefinition;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.CaseDetails;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.Event;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.Callback;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PostSubmitCallbackResponse;
+import uk.gov.hmcts.reform.iacaseapi.domain.handlers.PostSubmitCallbackHandler;
+
+@Component
+@Slf4j
+public class EditDocsAuditLogHandler implements PostSubmitCallbackHandler<AsylumCase> {
+
+    @Autowired
+    private UserDetailsProvider userDetailsProvider;
+
+    @Autowired
+    @Qualifier("hearingRecordingDocAudit")
+    private EditDocsAuditService editDocsAuditHearingRecordingDocTypeService;
+
+    @Autowired
+    @Qualifier("documentWithMetadataAudit")
+    private EditDocsAuditService documentWithMetadataAuditService;
+
+    @Override
+    public boolean canHandle(Callback<AsylumCase> callback) {
+        requireNonNull(callback, "callback must not be null");
+        return callback.getEvent() == Event.EDIT_DOCS;
+    }
+
+    @Override
+    public PostSubmitCallbackResponse handle(Callback<AsylumCase> callback) {
+        log.info("Edit Document audit logs...");
+        log.info("CCD case id: {}", callback.getCaseDetails().getId());
+        log.info("Delete/Update document ids: {}", getDeletedDocIds(callback.getCaseDetails(),
+            callback.getCaseDetailsBefore().orElse(null)));
+        log.info("IDAM User id: {}", userDetailsProvider.getUserDetails().getId());
```

Comment:
maybe I am wrong, but I think there is no `EDIT_DOCS_REASON` captured from case data

Reply:
Thanks for your comment @makowalski . Edit docs reason does not need to be audited in the logs. It is stored against the case in CCD as a casenote, this is the req: https://tools.hmcts.net/jira/browse/RIA-2831

Reply:
:+1: